chatsky/responses/rag.py

```diff
@@ -0,0 +1,62 @@
+from typing import Dict, Any
+from sentence_transformers import SentenceTransformer
+from qdrant_client import QdrantClient
+from chatsky.responses.base_response import BaseResponse
+
+
+class RAGPromptBuilder(BaseResponse):
+    def __init__(self, vector_store_url: str, retriever_model: str, **kwargs):
+        super().__init__(**kwargs)
+        self.vector_store = QdrantClient(url=vector_store_url)
+        self.retriever = SentenceTransformer(retriever_model)
+
+    def execute(self, user_query: str, ctx: Dict[str, Any], **kwargs) -> Dict[str, Any]:
+        if "cached_rag_context" in ctx:
+            context = ctx["cached_rag_context"]
+        else:
+            query_embedding = self.retriever.encode(user_query)
+            results = self.vector_store.search(
+                collection_name="docs",
+                query_vector=query_embedding,
+                limit=3
+            )
+            context = "\n".join([hit.payload["text"] for hit in results])
+            ctx["cached_rag_context"] = context
+
+        prompt = f"Контекст: {context}\nВопрос: {user_query}"
```

Comment:
Would like prompt customization (i.e. ability to change prompt template via `__init__`).